docs/source/conf.py

```diff
@@ -49,3 +50,8 @@
 # -- Options for extensions --------------------------------------------------
 
 autosummary_generate = True
+
+# ensure links to third-party docs work
+intersphinx_mapping = {
+    "sklearn": ("https://scikit-learn.org/stable/", None),
+}
```

Comment:
This fixes warnings like this:

```text
docstring of sklearn.utils._metadata_requests.RequestMethod.__get__.<locals>.func:3: WARNING: undefined label: 'metadata_routing'
```

Context for this change:

* https://github.com/microsoft/LightGBM/issues/5954
* https://github.com/microsoft/LightGBM/pull/5956